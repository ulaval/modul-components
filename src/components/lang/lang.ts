import Vue from 'vue';
import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './lang.html';
import { I18N_NAME } from '../component-names';
import { ModulVue } from '../../utils/vue';

@WithRender
@Component
export class MLang extends ModulVue {
    public text: string = '';

    @Prop()
    public msg: string;

    public mounted() {
        this.text = this.$i18n.translate(this.msg);
    }
}

const LangPlugin: PluginObject<any> = {
    install(v, options) {
        v.component(I18N_NAME, MLang);
    }
};

export default LangPlugin;
