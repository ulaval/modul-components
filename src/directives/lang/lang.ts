import Vue, { VNodeDirective } from 'vue';
import { PluginObject } from 'vue';
import { I18N_NAME } from '../directives-names';
import Messages from '@/utils/i18n';

export class MI18n extends Vue {
    public bind(element: HTMLElement, binding: VNodeDirective) {
        element.innerText = Messages.translate(binding.value);
    }
}

const MI18nPlugin: PluginObject<any> = {
    install(v, options) {
        v.directive(I18N_NAME, new MI18n());
    }
};

export default MI18nPlugin;
