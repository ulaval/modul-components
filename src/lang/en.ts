import { PluginObject } from 'vue';
import { BundleMessagesMap, ENGLISH, Messages } from '../utils/i18n/i18n';

const EnglishPlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug('EnglishPlugin', 'plugin.install');
        const i18n: Messages = (v.prototype).$i18n;
        if (i18n) {
            const msgs: BundleMessagesMap[] = [
                require('../components/accordion/accordion.lang.en.json'),
                require('../components/accordion-group/accordion-group.lang.en.json'),
                require('../components/list-item/list-item.lang.en.json'),
                require('../components/modal/modal.lang.en.json'),
                require('../components/chip/chip.lang.en.json'),
                require('../components/datefields/datefields.lang.en.json'),
                require('../components/datepicker/datepicker.lang.en.json'),
                require('../components/daterangepicker/daterangepicker.lang.en.json'),
                require('../components/dropdown/dropdown.lang.en.json'),
                require('../components/dropdown-group/dropdown-group.lang.en.json'),
                require('../components/overlay/overlay.lang.en.json'),
                require('../components/file-select/file-select.lang.en.json'),
                require('../components/file-upload/file-upload.lang.en.json'),
                require('../components/icon-file/icon-file.lang.en.json'),
                require('../components/inplace-edit/inplace-edit.lang.en.json'),
                require('../components/input-style/input-style.lang.en.json'),
                require('../components/link/link.lang.en.json'),
                require('../components/limit-text/limit-text.lang.en.json'),
                require('../components/login/login.lang.en.json'),
                require('../components/message/message.lang.en.json'),
                require('../components/menu/menu.lang.en.json'),
                require('../components/option/option.lang.en.json'),
                require('../components/dialog/dialog.lang.en.json'),
                require('../components/pagination/pagination.lang.en.json'),
                require('../components/phone-number/phone-number.lang.en.json'),
                require('../components/repeater/repeater.lang.en.json'),
                require('../components/rich-text-editor/rich-text-editor.lang.en.json'),
                require('../components/table/table.lang.en.json'),
                require('../components/toast/toast.lang.en.json'),
                require('../components/tree/tree.lang.en.json'),
                require('../components/scroll-top/scroll-top.lang.en.json'),
                require('../components/show-more/show-more.lang.en.json'),
                require('../components/sidebar/sidebar.lang.en.json'),
                require('../components/spinner/spinner.lang.en.json'),
                require('../components/switch/switch.lang.en.json'),
                require('../components/textfield/textfield.lang.en.json'),
                require('../components/timepicker/timepicker.lang.en.json'),
                require('../components/tooltip/tooltip.lang.en.json'),
                require('../components/validation-message/validation-message.lang.en.json'),
                require('../filters/filesize/filesize.lang.en.json'),
                require('../filters/date/date/date.lang.en.json'),
                require('../filters/date/period/period.lang.en.json'),
                require('../filters/date/time/time.lang.en.json'),
                require('../filters/date/date-time/date-time.lang.en.json')
            ];

            msgs.forEach(msgs => i18n.addMessages(ENGLISH, msgs));
        } else {
            throw new Error(
                'EnglishPlugin.install -> You must use the i18n plugin.'
            );
        }
    }
};

export default EnglishPlugin;
