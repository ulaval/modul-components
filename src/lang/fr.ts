import { PluginObject } from 'vue';

import { BundleMessagesMap, FRENCH, Messages } from '../utils/i18n/i18n';

const FrenchPlugin: PluginObject<any> = {
    install(v, options): void {
        console.debug('FrenchPlugin', 'plugin.install');
        const i18n: Messages = (v.prototype as any).$i18n;
        if (i18n) {
            const msgs: BundleMessagesMap[] = [
                require('../components/accordion/accordion.lang.fr.json'),
                require('../components/accordion-group/accordion-group.lang.fr.json'),
                require('../components/list-item/list-item.lang.fr.json'),
                require('../components/dialog/dialog.lang.fr.json'),
                require('../components/datefields/datefields.lang.fr.json'),
                require('../components/datepicker/datepicker.lang.fr.json'),
                require('../components/dropdown/dropdown.lang.fr.json'),
                require('../components/dropdown-item/dropdown-item.lang.fr.json'),
                require('../components/dropdown-group/dropdown-group.lang.fr.json'),
                require('../components/edit-window/edit-window.lang.fr.json'),
                require('../components/file-select/file-select.lang.fr.json'),
                require('../components/file-upload/file-upload.lang.fr.json'),
                require('../components/input-style/input-style.lang.fr.json'),
                require('../components/link/link.lang.fr.json'),
                require('../components/limit-text/limit-text.lang.fr.json'),
                require('../components/message/message.lang.fr.json'),
                require('../components/menu/menu.lang.fr.json'),
                require('../components/modal/modal.lang.fr.json'),
                require('../components/phone-number/phone-number.lang.fr.json'),
                require('../components/scroll-top/scroll-top.lang.fr.json'),
                require('../components/sidebar/sidebar.lang.fr.json'),
                require('../components/spinner/spinner.lang.fr.json'),
                require('../components/switch/switch.lang.fr.json'),
                require('../components/textfield/textfield.lang.fr.json'),
                require('../components/timepicker/timepicker.lang.fr.json'),
                require('../components/tooltip/tooltip.lang.fr.json'),
                require('../components/validation-message/validation-message.lang.fr.json')
            ];

            msgs.forEach(msgs => i18n.addMessages(FRENCH, msgs));
        } else {
            throw new Error(
                'FrenchPlugin.install -> You must use the i18n plugin.'
            );
        }
    }
};

export default FrenchPlugin;
