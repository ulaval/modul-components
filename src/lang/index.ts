/**
 * List public declaration here.
 */
export * from './fr';
