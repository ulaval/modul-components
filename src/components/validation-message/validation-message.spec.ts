import { createLocalVue, mount } from '@vue/test-utils';
import Vue, { VueConstructor } from 'vue';

import { addMessages } from '../../../tests/helpers/lang';
import { renderComponent } from '../../../tests/helpers/render';
import ValidationMessagePlugin, { MValidationMessage } from '../validation-message/validation-message';

describe('MValidationMessage', () => {
    let localVue: VueConstructor<Vue>;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(ValidationMessagePlugin);
        addMessages(localVue, [
            'components/validation-message/validation-message.lang.en.json'
        ]);
    });

    it('should render nothing when there is no validation message', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue
        });

        expect(renderComponent(valMsg.vm)).resolves.toEqual('');
    });

    it('should render nothing when it is disabled', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                disabled: true
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toEqual('');
    });

    it('should render nothing when it is waiting', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                waiting: true
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toEqual('');
    });

    it('should render nothing when it is disabled', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                disabled: true
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toEqual('');
    });

    it('should render helper message when prop is set', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                helperMessage: 'help'
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when it is valid', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                validMessage: 'valid'
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when it is in error', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                errorMessage: 'error'
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when it is in error', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                errorMessage: 'error'
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toMatchSnapshot();
    });

    it('should render error message even if there is a valid message', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                validMessage: 'valid',
                errorMessage: 'error'
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toMatchSnapshot();
    });

    it('should emit click event when clicked', () => {
        const valMsg = mount(MValidationMessage, {
            localVue: localVue,
            propsData: {
                validMessage: 'valid',
                errorMessage: 'error'
            }
        });

        valMsg.find('div').trigger('click');

        expect(valMsg.emitted('click')).toBeTruthy();
    });

    /*
    it('should render correctly when position prop is right', () => {
        const valMsg = mount(MCheckbox, {
            localVue: localVue,
            propsData: {
                position: MCheckboxPosition.Right
            }
        });

        expect(renderComponent(valMsg.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when value prop is true', () => {
        const chkbox = mount(MCheckbox, {
            localVue: localVue,
            propsData: {
                value: true
            }
        });

        expect(renderComponent(chkbox.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when disabled', () => {
        const chkbox = mount(MCheckbox, {
            localVue: localVue,
            propsData: {
                disabled: true
            }
        });

        expect(renderComponent(chkbox.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly when a label is provided', () => {
        const chkbox = mount(MCheckbox, {
            localVue: localVue,
            slots: { default: 'label' }
        });

        expect(renderComponent(chkbox.vm)).resolves.toMatchSnapshot();
    });

    it('should emit click event when clicked', () => {
        const chkbox = mount(MCheckbox, {
            localVue: localVue
        });

        chkbox.find('input').trigger('click');

        expect(chkbox.emitted('click')).toBeTruthy();
    });

    it('should flow down InputState mixin props to m-validation-message', () => {
        const valMsgProps = {
            disabled: false,
            error: false,
            errorMessage: 'error-message',
            validMessage: 'valid-message',
            helperMessage: 'helper-message'
        };

        const chkbox = mount(MCheckbox, {
            localVue: localVue,
            propsData: valMsgProps,
            computed: {
                hasError() {
                    return false;
                },
                isDisabled() {
                    return false;
                }
            }
        });

        expect(chkbox.find(MValidationMessage).props()).toEqual(valMsgProps);
    }); */
});
