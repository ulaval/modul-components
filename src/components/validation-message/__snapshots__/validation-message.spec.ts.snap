// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MValidationMessage should render correctly when it is in error 1`] = `
<div class="m-validation-message">
    <p class="m-validation-message__error">
        <svg width="1em" height="1em" class="m-icon m-validation-message__error__icon">
            <title>Error</title>
            <use xlink:href="#error" aria-hidden="true"></use>
        </svg><span class="m-validation-message__text">error</span></p>
</div>
`;

exports[`MValidationMessage should render correctly when it is in error 2`] = `
<div class="m-validation-message">
    <p class="m-validation-message__error">
        <svg width="1em" height="1em" class="m-icon m-validation-message__error__icon">
            <title>Error</title>
            <use xlink:href="#error" aria-hidden="true"></use>
        </svg><span class="m-validation-message__text">error</span></p>
</div>
`;

exports[`MValidationMessage should render correctly when it is valid 1`] = `
<div class="m-validation-message">
    <p class="m-validation-message__valid">
        <svg width="1em" height="1em" class="m-icon m-validation-message__valid__icon">
            <title>Valid</title>
            <use xlink:href="#check" aria-hidden="true"></use>
        </svg> <span class="m-validation-message__text">valid</span></p>
</div>
`;

exports[`MValidationMessage should render error message even if there is a valid message 1`] = `
<div class="m-validation-message">
    <p class="m-validation-message__error">
        <svg width="1em" height="1em" class="m-icon m-validation-message__error__icon">
            <title>Error</title>
            <use xlink:href="#error" aria-hidden="true"></use>
        </svg><span class="m-validation-message__text">error</span></p>
</div>
`;

exports[`MValidationMessage should render helper message when prop is set 1`] = `
<div class="m-validation-message">
    <p class="m-validation-message__helper">help</p>
</div>
`;
